common/src/gadget/mod.rs

```diff
@@ -1,31 +1,35 @@
-use crate::client::{ClientWithApi, JobsClient};
+use crate::client::JobsClient;
 use crate::debug_logger::DebugLogger;
-use crate::gadget::message::GadgetProtocolMessage;
-use crate::gadget::work_manager::WorkManager;
+use crate::environments::GadgetEnvironment;
+use crate::gadget::tangle::JobInitMetadata;
 use crate::protocol::{AsyncProtocol, AsyncProtocolRemote};
 use crate::tangle_runtime::*;
 use crate::Error;
 use async_trait::async_trait;
-use gadget_core::gadget::substrate::{FinalityNotification, SubstrateGadgetModule};
+use gadget_core::gadget::general::Client;
+use gadget_core::gadget::general::GadgetWithClient;
+use gadget_core::gadget::manager::AbstractGadget;
 use gadget_core::job::{BuiltExecutableJobWrapper, ExecutableJob, JobBuilder};
-use gadget_core::job_manager::{PollMethod, ProtocolWorkManager, WorkManagerInterface};
+use gadget_core::job_manager::{ProtocolWorkManager, WorkManagerInterface};
 use network::Network;
 use parking_lot::{Mutex, RwLock};
-use sp_core::{ecdsa, keccak_256, sr25519};
+use sp_core::sr25519;
 use std::sync::Arc;
 use std::time::Duration;
 
+pub mod core;
 pub mod message;
 pub mod network;
+pub mod tangle;
 pub mod work_manager;
 
 /// Used as a module to place inside the SubstrateGadget
-pub struct Module<C, N, M> {
+pub struct GeneralModule<N, M, Env: GadgetEnvironment> {
```

Comment:
I think `Module` is better for succinctness if you're just putting the word `General` on it. That doesn't describe it any more than keeping it as is.